### ISSUE TITLE
Ambiguous name of galois::no_conflicts

### ISSUE THREAD

**Issue by Michael-JY-He** [IntelligentSoftwareSystems/Galois#260]
A user puts galois::no_conflicts to avoid conflicts. However this actually disables conflict checking and results in conflicts.

Should change the name.

**Comment by l-hoang** [comment]
It's a subjective matter: the way I see it, specifying no_conflicts is like telling the loop "there will be no conflicts in this loop, therefore disable conflict checking for better performance". I can see how it can be thought of as "I don't want conflicts if I specify no_conflicts", though.

I'm personally fine with the way it is right now.

@amberhassaan @insertinterestingnamehere @ddn0 thoughts?

**Comment by ddn0** [comment]
Yeah the original intent was that these loop traits would be assertions about the user code that we couldn't automatically determine but a user could assert (e.g., no conflicts, needs pushes, etc.) and the system would take advantage off ("oh your code doesn't have conflicts, then I'll disable conflict detection"). Though I can understand the confusion because some other traits like `steal` are named differently and they imply that "the system should do X" rather than "X is a property of my code".

**Comment by roshandathathri** [comment]
Yes, we should have the same meaning for all traits. They should either be "the system should do X" or "X is a property of my code". I tend towards the former, although I'm fine with the latter. Specifying the presence or lack of properties can be misleading, but specifying doX and don'tDoX is more straightforward in my opinion as it is similar to enabling a flag (and that do or don't will be in the flag name).

**Comment by insertinterestingnamehere** [comment]
I see the confusion, though the current naming makes more sense to me. Maybe we can name it "check_conflicts", or something similar?

**Comment by roshandathathri** [comment]
Checking conflicts should be the default, so it should dont_check_conflicts or disable_conflict_checks.

**Comment by bozhiyou** [comment]
Personal thought: why not just introduce one more namespace to make it something like `galois::assertion::no_conflicts()` or `galois::action::steal()`? I think it's enough for disambiguation.

**Comment by ddn0** [comment]
I think it is not common to have more than one or two namespaces in a C++ codebase compared to other languages like Python, Java, C#, etc. Here's an article about some of the issues that can arise: https://abseil.io/tips/130

In the case of `galois::no_conflicts`, we probably don't want to inherit all the additional meaning that adding a C++ namespace implies, so I think just changing the actual name would be more appropriate here.

**Comment by bozhiyou** [comment]
Good resource to read. I agree. Thanks!

**Comment by roshandathathri** [comment]
Is everyone fine with galois::disable_conflict_detection()?

**Comment by l-hoang** [comment]
Slightly too wordy for my liking but it works + is descriptive enough. If I come up with something else before you get to this I'll let you know.

**Comment by bozhiyou** [comment]
Just noticed this in the tutorial: should `no_pushes` also be `disable_context_push` or something for better consistency?

**Comment by roshandathathri** [comment]
That is a property, not an action. We should make it consistent later/separately. I'm not sure if everything can be an action. In this case, I think it's better to has_no_pushes to indicate that it is property.

**Comment by ddn0** [comment]
Since the current issue is closed I'd recommend continuing the discussion elsewhere. I created #280 for follow-on work.
